lib/galaxy/datatypes/proteomics.py

```diff
@@ -0,0 +1,396 @@
+    root = "protein_summary"
+
+
+class MzXML(ProteomicsXml):
+    """mzXML data"""
+    file_ext = "mzxml"
+    blurb = "mzXML Mass Spectrometry data"
+    root = "mzXML"
+
+
+class MzIdentML(ProteomicsXml):
+    file_ext = "mzid"
+    edam_format = "format_3247"
+    blurb = "XML identified peptides and proteins."
+    root = "MzIdentML"
+
+
+class TraML(ProteomicsXml):
+    file_ext = "traml"
+    edam_format = "format_3246"
+    blurb = "TraML transition list"
+    root = "TraML"
+
+
+class MzQuantML(ProteomicsXml):
+    file_ext = "mzq"
+    edam_format = "format_3248"
+    blurb = "XML quantification data"
+    root = "MzQuantML"
+
+
+class ConsensusXML(ProteomicsXml):
+    file_ext = "consensusxml"
+    blurb = "OpenMS multiple LC-MS map alignment file"
+    root = "consensusXML"
+
+
+class FeatureXML(ProteomicsXml):
+    file_ext = "featurexml"
+    blurb = "OpenMS feature file"
+    root = "featureMap"
+
+
+class IdXML(ProteomicsXml):
+    file_ext = "idxml"
+    blurb = "OpenMS identification file"
+    root = "IdXML"
+
+
+class TandemXML(ProteomicsXml):
+    file_ext = "tandem"
+    blurb = "X!Tandem search results file"
+    root = "bioml"
+
+
+class Mgf(Text):
+    """Mascot Generic Format data"""
+    file_ext = "mgf"
+
+    def set_peek(self, dataset, is_multi_byte=False):
+        """Set the peek and blurb text"""
+        if not dataset.dataset.purged:
+            dataset.peek = data.get_file_peek(dataset.file_name, is_multi_byte=is_multi_byte)
+            dataset.blurb = 'mgf Mascot Generic Format'
+        else:
+            dataset.peek = 'file does not exist'
+            dataset.blurb = 'file purged from disk'
+
+    def sniff(self, filename):
+        mgf_begin_ions = "BEGIN IONS"
+        max_lines = 100
+
+        with open(filename) as handle:
+            for i, line in enumerate(handle):
+                line = line.rstrip()
+                if line == mgf_begin_ions:
+                    return True
+                if i > max_lines:
+                    return False
+
+
+class MascotDat(Text):
+    """Mascot search results """
+    file_ext = "mascotdat"
+
+    def set_peek(self, dataset, is_multi_byte=False):
+        """Set the peek and blurb text"""
+        if not dataset.dataset.purged:
+            dataset.peek = data.get_file_peek(dataset.file_name, is_multi_byte=is_multi_byte)
+            dataset.blurb = 'mascotdat Mascot Search Results'
+        else:
+            dataset.peek = 'file does not exist'
+            dataset.blurb = 'file purged from disk'
+
+    def sniff(self, filename):
+        mime_version = "MIME-Version: 1.0 (Generated by Mascot version 1.0)"
+        max_lines = 10
+
+        with open(filename) as handle:
+            for i, line in enumerate(handle):
+                line = line.rstrip()
+                if line == mime_version:
+                    return True
+                if i > max_lines:
+                    return False
+
+
+class RAW(Binary):
```

Comment:
@iracooke do you think we should rename this class to something like `ThermoFinniganRaw`? `RAW` seems really general.